src/rade_api.c

```diff
@@ -0,0 +1,315 @@
+
+    return 0;
+}
+
+void rade_tx_close(struct rade *r) {
+  // TODO we may need more of these, see if there are any memory leaks
+  Py_DECREF(r->pArgs_radae_tx);
+  Py_DECREF(r->pFunc_radae_tx);
+  Py_DECREF(r->pFunc_radae_tx_eoo);
+  Py_DECREF(r->pArgs_radae_tx_eoo);
+  Py_DECREF(r->pModule_radae_tx);
+
+  free(r->features_in);
+  free(r->tx_out);
+  free(r->tx_eoo_out);
+}
+
+// returns 0 for success
+int rade_rx_open(struct rade *r) {
+    PyObject *pName, *pClass;
+    PyObject *pValue;
+    PyObject *pArgs;
+    char *python_module_name = "radae_rx";
+    char *do_radae_rx_meth_name = "do_radae_rx";
+
+    // Load module of Python code
+    pName = PyUnicode_DecodeFSDefault(python_module_name);
+    r->pModule_radae_rx = PyImport_Import(pName);
+    check_error(r->pModule_radae_rx, "importing", python_module_name);
+    Py_DECREF(pName);
+
+    // Find class and create an instance
+    pClass = PyObject_GetAttrString(r->pModule_radae_rx, "radae_rx");
+    check_error(pClass, "finding class", "radae_rx");
+    pArgs = Py_BuildValue("(s)", "../model19_check3/checkpoints/checkpoint_epoch_100.pth");
+    r->pInst_radae_rx = PyObject_CallObject(pClass, pArgs);
+    check_error(r->pInst_radae_rx, "Creating instance of class", "radae_rx");
+    Py_DECREF(pClass);
+    Py_DECREF(pArgs);
+
+    r->n_features_out = (int)call_getter(r->pInst_radae_rx, "get_n_features_out");
+    r->nin_max = (int)call_getter(r->pInst_radae_rx, "get_nin_max");
+    r->nin = (int)call_getter(r->pInst_radae_rx, "get_nin");
+    fprintf(stderr, "n_features_out: %d nin_max: %d nin: %d\n", (int)r->n_features_out, (int)r->nin_max, (int)r->nin);
+        
+    r->pMeth_radae_rx = PyObject_GetAttrString(r->pInst_radae_rx, do_radae_rx_meth_name);
+    check_error(r->pMeth_radae_rx, "finding",  do_radae_rx_meth_name);
+    check_callable(r->pMeth_radae_rx, do_radae_rx_meth_name, "not callable");
+
+    r->pArgs_radae_rx = PyTuple_New(2);
+
+    // 1st Python function arg - input numpy array of csingle rx samples
+    r->rx_in = (RADE_COMP*)malloc(sizeof(RADE_COMP)*r->nin_max);
+    assert(r->rx_in != NULL);
+    pValue = PyArray_SimpleNewFromData(1, &r->nin_max, NPY_CFLOAT, r->rx_in);
+    check_error(pValue, "setting up numpy array", "buffer_complex");
+    PyTuple_SetItem(r->pArgs_radae_rx, 0, pValue);
+
+    // 2nd Python arg - output numpy array of float features
+    r->features_out = (float*)malloc(sizeof(float)*r->n_features_out);
+    assert(r->features_out != NULL);
+    pValue = PyArray_SimpleNewFromData(1, &r->n_features_out, NPY_FLOAT, r->features_out);
+    check_error(pValue, "setting up numpy array", "features_out");
+    PyTuple_SetItem(r->pArgs_radae_rx, 1, pValue);
+ 
+    return 0;
+}
+
+void rade_rx_close(struct rade *r) {
+  Py_DECREF(r->pArgs_radae_rx);
+  Py_DECREF(r->pMeth_radae_rx);
+  Py_DECREF(r->pInst_radae_rx);
+  Py_DECREF(r->pModule_radae_rx);
+
+  free(r->features_out);
+  free(r->rx_in);
+}
+
+struct rade *rade_open(char model_file[]) {
```

Comment:
`model_file` is being ignored here?

Reply:
Yep it's  placeholder for now ... but will be actively used soon.